apps/arweave/src/ar_chunk_storage.erl

```diff
@@ -389,7 +389,8 @@ handle_cast(do_prepare_replica_2_9, State) ->
 				%% storage modules are commonly set up with.
 				PaddedEndOffset2 = shift_replica_2_9_entropy_offset(
 					PaddedEndOffset, -SliceIndex),
-				store_entropy(Entropies, PaddedEndOffset2, SubChunkStart, Partition,
+				PartitionEnd = (Partition + 1) * ?PARTITION_SIZE,
```

Comment:
We should pad PartitionEnd it because we compare it with the padded offset. We should do the same for the CheckRangeEnd check above considering the updates.

Reply:
Updated! https://github.com/ArweaveTeam/arweave/pull/680/commits/6811ceae2fa0632e5fe830b34ebb1fd85585af55